src/localkiwixserver.cpp

```diff
@@ -24,16 +25,30 @@ LocalKiwixServer::LocalKiwixServer(QWidget *parent) :
 
     connect(ui->KiwixServerButton, SIGNAL(clicked()), this, SLOT(runOrStopServer()));
     connect(ui->OpenInBrowserButton, SIGNAL(clicked()), this, SLOT(openInBrowser()));
-    connect(KiwixApp::instance()->getSettingsManager(), &SettingsManager::portChanged,
-            this, [=](int port) { m_port = port; });
     connect(ui->closeButton, &QPushButton::clicked, this, &LocalKiwixServer::close);
-    const QHostAddress &localhost = QHostAddress(QHostAddress::LocalHost);
-    for (const QHostAddress &address: QNetworkInterface::allAddresses()) {
-        if (address.protocol() == QAbstractSocket::IPv4Protocol && address != localhost) {
-            m_ipAddress = address.toString();
-            break;
+    connect(ui->PortChooser, &QLineEdit::textChanged, ui->PortChooser, [=](const QString &text) {
+        if (text.toInt() > 65535 || text.toInt() < 1) {
+            QString validText = text;
+            validText.chop(1);
+            ui->PortChooser->setText(validText);
         }
+    });
+
+    const auto interfacesMap = kiwix::getNetworkInterfaces();
+    QVector<QString> interfaces;
+    interfaces.reserve(interfacesMap.size() + 1);
+    for (const auto &interfacePair : interfacesMap) {
+        QString ip = QString::fromStdString(interfacePair.second);
+        interfaces.push_back(ip);
+    }
+    interfaces.push_back(QString("0.0.0.0"));
```

Comment:
I would rename it to "default" or "all".
If the user don't know about IP addresses and just want the "best" value, `0.0.0.0` doesn't help a lot.